In [1]:
import numpy as np

### Задача 1

#### Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм.
#### Используя уровень значимости 0.05 и правостороннюю критическую область, проверить данную гипотезу при условии,
#### что в выборке из 100 шариков средний диаметр оказался равным 17.5 мм, а дисперсия известна и равна 4 мм.

### Решение:

#### Cформулируем нулевую и альтернативную гипотезы.
#### Нулевая гипотеза H0: средний диаметр = 17, Альтернативная гипотеза (односторонняя) H1: средний диаметр > 17

#### Вычисляем наблюдаемое значение критерия:
#### ((средний диаметр в выборке - средний диаметр нулевой гипотезы)/дисперсия) * sqrt(кол-во шариков в выборке)

In [2]:
((17.5 - 17) / 4) * np.sqrt(100)

1.25

#### По таблице функции Лапласа найдем критическую точку для односторонней критической
#### области (при гипотезе H1 средний диаметр > 17 ) по уровню значимости = 0.05: 
#### (1 - 2 * уровень значимости) / 2

In [3]:
(1 - 2 * 0.05) / 2

0.45

#### критическая точка = 1.645

#### Так как 1.25 < 1.645, то нулевую гипотезу можно принять, следовательно можно считать,
#### что средний диаметр действительно = 17

### Задача 2

#### Известно, что генеральная совокупность распределена нормально со средним квадратическим отклонением 16.
#### Найти доверительный интервал для оценки математического ожидания a с надежностью 0.95,
#### если выборочное среднее равно 80, а объём равен 256.

### Решение:

#### Из условия 2 * Функция Лапласа = 0.95 находим Функцию Лапласа:

In [4]:
0.95 / 2

0.475

#### По таблице Функции Лапласса для 0.475 находим значение аргумента = 1.96

#### Точность оценки при среднем квадратическом отклонении = 16 и объеме выборки = 256:

In [5]:
1.96 * 16 / np.sqrt(256)

1.96

#### Доверительный интервал для оценки математического ожидания нормально распределенного количественного признака
#### по выборочной средней = 80 при известном среднем квадратическом отклонении = 16 генеральной совокупности:
#### выборочное среднее - точность оценки < a < выборочное среднее + точность оценки

In [6]:
80 - 1.96

78.04

In [7]:
80 + 1.96

81.96

#### 78.04 < a < 81.96

### Задача 3

#### Продавец утверждает, что средний вес пачки печенья составляет 200 г.
#### Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет:
#### 202, 203, 199, 197, 195, 201, 200, 204, 194, 190
#### Известно, что их веса распределены нормально.
#### Верно ли утверждение продавца, если учитывать, что доверительная вероятность равна 99%?

### Решение:

#### Для выборки из 10 пачек:

In [8]:
samples = np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])

#### посчитаем показатли выборки: выборочное среднее и выборочное исправленное среднее квадратическое отклонение

In [9]:
samples.mean()

198.5

In [10]:
samples.std(ddof=1)

4.453463071962462

#### Введем нулевую гипотезу H0: средний вес пачки равен 200 при конкурирующей гипотезе H1: средний вес пачки не равен 200. 

#### Вычислим наблюдаемое значение критерия:

In [11]:
((195.5 - 200) / 4.453463071962462) * np.sqrt(10)

-3.1953222112352693

#### По таблице критических точек распределения Стьюдента найдем критическую точку по
#### уровню значимости = 0.01 и числу степеней свободы = 9 , откуда критическая точка  = 3.25. 

#### Так как наблюдаемое значение критерия(по модулю) 3.1953222112352693 < критической точки = 3.25,
#### то нулевую гипотезу о среднем весе = 200 можно принять. То есть утверждение продавца верно.